miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -271,6 +270,11 @@ private Sample save(Sample sample) throws IOException {
         created.setAlias(created.getName());
         needsUpdate = true;
       }
+      if (autoGenerateIdBarcodes && isStringEmptyOrNull(created.getIdentificationBarcode())) {
```

Comment:
It can't happen before create because the name (generated after create) is used in the idBarcode